ts-src/lib/connection.ts

```diff
@@ -42,7 +42,7 @@ export function createConnection({
   const jt400: Connection = {
     ...baseConnection,
     async transaction(transactionFunction) {
-      const t = connection.createTransactionSync()
+      const t = await connection.createTransaction()
```

Comment:
How do you know this is a valid call??? I'm not sure node-java generates functions like that.

Reply:
java bridge has definitions for this!

<img width="859" alt="image" src="https://github.com/user-attachments/assets/6588ab8a-46b2-4108-9f03-8d84af635c8a">

notice the promiseSuffix is nothing meaning the base method is promisified

Reply:
i am convinced!